cargo-dist/templates/installer/homebrew.rb.j2

```diff
@@ -1,70 +1,32 @@
 class {{ formula_class }} < Formula
-  {%- if desc %}
   desc "{{ desc }}"
-  {%- endif %}
-  {%- if homepage %}
+  {%- if homepage is defined %}
   homepage "{{ homepage }}"
   {%- endif %}
   version "{{ inner.app_version }}"
-  {#- #}
-  {%- if arm64_macos.id or x86_64_macos.id %}
-  if OS.mac?
-    {#- If arm64/x86_64 builds are the same, skip the Hardware::CPU.type if statement #}
-    {%- if arm64_macos.id == x86_64_macos.id %}
-    url "{{ inner.base_url }}/{{ arm64_macos.id }}"
-    {%- if arm64_macos_sha256 %}
-    sha256 "{{ arm64_macos_sha256 }}"
-    {%- endif %}
-    {%- else %}
-    {%- if arm64_macos.id %}
-    if Hardware::CPU.arm?
-      url "{{ inner.base_url }}/{{ arm64_macos.id }}"
-      {%- if arm64_macos_sha256 %}
-      sha256 "{{ arm64_macos_sha256 }}"
-      {%- endif %}
-    end
-    {%- endif %}
-    {%- if x86_64_macos.id %}
-    if Hardware::CPU.intel?
-      url "{{ inner.base_url }}/{{ x86_64_macos.id }}"
-      {%- if x86_64_macos_sha256 %}
-      sha256 "{{ x86_64_macos_sha256 }}"
-      {%- endif %}
-    end
-    {%- endif %}
-    {%- endif %}
-  end
-  {%- endif %}
-  {#- #}
-  {%- if arm64_linux.id or x86_64_linux.id %}
-  if OS.linux?
-    {%- if arm64_linux.id %}
-    if Hardware::CPU.arm?
-      url "{{ inner.base_url }}/{{ arm64_linux.id }}"
-      {%- if arm64_linux %}
-      sha256 "{{ arm64_linux_sha256 }}"
-      {%- endif %}
-    end
-    {%- endif %}
-    {%- if x86_64_linux.id %}
-    if Hardware::CPU.intel?
-      url "{{ inner.base_url }}/{{ x86_64_linux.id }}"
-      {%- if x86_64_linux_sha256 %}
-      sha256 "{{ x86_64_linux_sha256 }}"
-      {%- endif %}
-    end
+
+  {% macro fragment_url_and_hash(cond, frag) -%}
+  {%- if frag %}
+  if {{ cond }}
+    url "{{ inner.base_url }}/{{ frag.id }}"
+    {%- if frag.sha256 is defined %}
+    sha256 "{{ frag.sha256 }}"
     {%- endif %}
-  end
+  endif
```

Comment:
Typo? This is in the Ruby code, not jinja syntax - Ruby uses `if/end`, not `if/endif`.
```suggestion
  end
```